### PR TITLE
Reload client CAs periodically

### PR DESCRIPTION
**Pull request by akorompai**
**Description:** Reload client CAs used for mTLS in sim

**Link to tracking Issue:**  #6524

**Testing:** manual only

**Documentation:** none

### CODE REVIEW

File: config/configtls/configtls.go [hunk 1]
```diff
@@ -229,18 +229,82 @@ func (c TLSClientSetting) LoadTLSConfig() (*tls.Config, error) {
...
+type clientCAsReloader struct {
+	// Path to the client CAs
+	ClientCAsFile string
+	// ReloadInterval specifies the duration after which the certificate will be reloaded
+	// If not set, it will never be reloaded (optional)
+	ReloadInterval time.Duration
+	nextReload     time.Time
+	certPool       *x509.CertPool
+	lock           sync.RWMutex
+	c              *TLSServerSetting
```
Comment (bogdandrutu) [inline config/configtls/configtls.go:244]: We already have a reload mechanism, what does this bring?

Comment (bogdandrutu) [reply config/configtls/configtls.go:244]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/configtls.go#L105

Comment (akorompai) [reply config/configtls/configtls.go:244]: Hi @bogdandrutu ! 

I'll gladly admit that this is a blatant copy of `certReloader`, but I found that the reloading of `client_ca_file` and `cert_file/key_file` are different enough concerns to have their own reloaders.
One first one is only related to server, the other one is related to both client and server, their lifecycles differ, etc...
Please let me know what would be a better approach!

Comment (akorompai) [reply config/configtls/configtls.go:244]: @bogdandrutu  meanwhile can you allow the CI to run, that would be a great feedback for us!

**Comment by jpkrohling** [comment]
Are certs always local files at the moment? If so, why not just watch the file and reload on that event, instead of reloading it periodically?

**Comment by akorompai** [comment]
That would work for us too, this PR extend the existing pattern to reload client CAs.
Would that have a better chance to be accepted?

**Comment by jpkrohling** [comment]
> Would that have a better chance to be accepted?

I would certainly prefer watching files to running a process periodically. We have this pattern elsewhere as well. I believe @frzifus implemented this recently for Jaeger's remote sampling extension.

**Comment by frzifus** [comment]
We implemented something to reload on change here: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.66.0/extension/bearertokenauthextension/bearertokenauth.go#L101-L134

**Comment by akorompai** [comment]
Could you check #6708 ? tried to adapt @frzifus 's solution.

**Comment by akorompai** [comment]
closing in favor of #6708

**Pull request closed without merging**